includes/model/class-follower.php

```diff
@@ -205,25 +215,38 @@ public function from_meta( $meta ) {
 	 * @return mixed The attribute value.
 	 */
 	public function get( $attribute ) {
-		if ( $this->$attribute ) {
+		if ( ! is_null( $this->$attribute ) ) {
 			return $this->$attribute;
 		}
-
-		$attribute = get_term_meta( $this->id, $attribute, true );
-		if ( $attribute ) {
-			$this->$attribute = $attribute;
-			return $attribute;
+		$attribute_value = get_term_meta( $this->id, $attribute, true );
+		if ( $attribute_value ) {
+			$this->$attribute = $attribute_value;
+			return $attribute_value;
 		}
 
-		$attribute = $this->get_meta_by( $attribute );
-		if ( $attribute ) {
-			$this->$attribute = $attribute;
-			return $attribute;
+		$attribute_value = $this->get_meta_by( $attribute );
+		if ( $attribute_value ) {
+			$this->$attribute = $attribute_value;
+			return $attribute_value;
 		}
 
 		return null;
 	}
 
+	/**
+	 * Get a URL for the follower. Creates one out of the actor if no URL was set.
+	 */
+	public function get_url() {
+		if ( $this->get( 'url' ) ) {
+			return $this->get( 'url' );
+		}
+		$actor = $this->get_actor();
+		// normalize
+		$actor = ltrim( $actor, '@' );
+		$parts = explode( '@', $actor );
+		return sprintf( 'https://%s/@%s', $parts[1], $parts[0] );
```

Comment:
This will not work for every platform! If you have a WebFinger-ID and you want to have a URL, you have to use a WebFinger Call to get the Profile-URL.

Reply:
The actor is a URL in most of the cases btw. I am not aware of a platform that sends WebFinger IDs as Actor. So we should check if the Actor is already a URL and if not, get a URL using a WebFinger call.

Reply:
I just checked the followers of my blog and all 500++ are having URLs as Actors.

Reply:
I would simply change the code to something like

```php
if ( filter_var( $this->actor, FILTER_VALIDATE_URL ) )
    $this->url = $this->actor;
} else {
    $this->url = Webfinger::resolve( $this->actor );

```

Reply:
Thanks! This is what comes from doing a lot of manual testing locally.